### PR TITLE
Added functions

### PR DESCRIPTION
**Pull request by stangier**
I added the function to show a custom label in front of the hard drive monitor:

asd

This is called by:

``` ruby
@manager.add_block Barr::Blocks::Hdd.new(device: "sda6", name: "/root")`
```

### CODE REVIEW

**Comment by stangier** [comment]
I added the class 'processes' which shows the number of active processes.
Called by:

``` ruby
@manager.add_block Barr::Blocks::Processes.new(bcolor: "#FFAAAA'", fcolor:"#333333")
```

File: lib/barr/blocks/hdd.rb [hunk 1]
```diff
@@ -4,12 +4,13 @@ class Hdd < Block
...
       def initialize opts={}
         super
+	@name = opts[:name]
         @device = opts[:device]
       end
 
       def update
         total, used, perc = sys_cmd.chomp.split(" ")
-        @output = "#{used} / #{total} (#{perc})"                                                              
+        @output = "#{@name}: #{used} / #{total} (#{perc})"
```
Comment (OkayDave) [inline lib/barr/blocks/hdd.rb:13]: Nice idea, though the functionality for this is already provided via the `icon` option.

`Barr::Blocks::Hdd.new device: "sda2", icon: "/home: "`

File: lib/barr/blocks/processes.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  module Blocks
+
+    class Processes < Block
+      def initialize opts={}
+        super 
+        @output = sys_cmd
+      end
+
+      def sys_cmd
+        `ps -e | wc -l`
```
Comment (OkayDave) [inline lib/barr/blocks/processes.rb:14]: Could you call `chomp` after the system cmd please? It's a minor detail as the manager will also remove any newlines as a safety measure, but I'd rather the blocks sanitised their own outputs as much as possible.

**Comment by OkayDave** [comment]
Thanks for the PR @derlorenz! 

I've added a couple of in-line comments. I think your `Processes` block would make a great addition. Would you mind writing some specs for it please, and adding documentation for it to the README.

If you're not familiar with specs, have a look at the `/spec/blocks/` folder for examples.

**Comment by stangier** [comment]
I just added the battery class which shows the batter percentage for laptops. You can pass a true or false argument to set if the remaining time should be shown or not. Also I fixed the process class not updating because it did not have an update function.

I will write specs for it and also an entry to the README.

**Comment by OkayDave** [comment]
@derlorenz thanks for the updates. I've made a few changes to your code to make it more in-line with the projects coding style, and to update it to the newer format. I'm closing your PR and I've opened a new one here: https://github.com/OkayDave/barr/pull/12/files.

Before I can merge it into the main branch, I need to update the Battery spec with a mocked output. Mocking and Stubbing API and system calls is really important as it ensures the specs don't fail due to an API or command being unavailable. In my case, I don't use a laptop so I'm unable to run the specs for the `acpi` call.

Could you post the output of `acpi | cut -d ',' -f 2-3` and `acpi | cut -d ',' -f 2` from your terminal please? I'll then add it to the specs and get your work merged in.

**Comment by stangier** [comment]
``` bash
acpi | cut -d "," -f 2-3
```

 96%, 03:17:59 remaining

``` bash
acpi | cut -d "," -f 2
```

 96%

**Pull request closed without merging**